server/index.js

```diff
@@ -35,32 +35,70 @@ app.get('/listings', function (req, res) {
       res.sendStatus(500);
     } else {
       res.json(data);
-      console.log(data);
+    }
+  });
+});
+
+
+app.post('/listings', function (req, res) {
+  //expects req.body to be an object with keys that include id, type, category, price, avgReview, numReview, numBeds, title, isFav
+
+  db.insertListing(req.body, function(err, results) {
+    if (err) {
+      res.sendStatus(500);
+    } else {
+      res.sendStatus(201);
+    }
+    res.end();
+  });
+});
 
+app.put('/listings', function(req, res) {
+  let queryParams = [];
+  for (let each in req.body) {
+    queryParams.push(req.body[each]);
+  }
+  console.log('req.body', req.body);
+  db.updateListing(queryParams, (err, results) => {
+    if (err) {
+      console.log('error', err);
+      res.sendStatus(500);
+    } else {
+      console.log('res', results);
+      res.sendStatus(202);
     }
+    res.end();
   });
+
 });
 
+app.delete('/listings', function(req, res) {
+  db.deleteListing([req.query.id], function(err, data) {
+    if (err) {
+      res.sendStatus(500);
+    } else {
+      res.sendStatus(202);
+    }
+    res.end();
+  });
+});
 
-app.get('/images', urlencodedParser, function (req, res) {
-  console.log('server /images');
+app.get('/images', function (req, res) {
   db.selectImages(req.body, function(err, data) {
     if (err) {
       res.sendStatus(500);
     } else {
       res.json(data);
```

Comment:
this one too, you can send a status of success along with res.json(data). 
res.status(200).json(data)